perf/src/recycler.rs

```diff
@@ -74,12 +75,19 @@ impl<T: Default + Reset + Sized> Recycler<T> {
     }
 
     pub fn allocate(&self, name: &'static str) -> T {
-        let new = self
-            .recycler
-            .gc
-            .lock()
-            .expect("recycler lock in pb fn allocate")
-            .pop();
+        let new = {
+            let mut gc = self
+                .recycler
+                .gc
+                .lock()
+                .expect("recycler lock in pb fn allocate");
+            if let Some((oldest_time, _old_item)) = gc.front() {
```

Comment:
Thought about maybe checking the length of `gc` here. Maybe the recycler is just not used in a while and we are popping an extra one where we only have a handful of entries (1-5).

Reply:
yeah, nice addition! Also, I noticed we shouldn't pop and drop the last one. We can just recycle it.. ;) I chose 10 for no particular strong reason. I just felt expiring with 5 items sounded too aggressive.... just that.

Reply:
yea, sounds good.